### PR TITLE
BTCORE-715 - align fork with origin

### PR DESCRIPTION
**Pull request by denys-klymenko-sigma**
* upgrade of zetasql dependency
* merge some improvements from original repo

With this update comes support for IFNULL and other query options that we use

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -749,7 +864,7 @@ func TestDuplicateTable(t *testing.T) {
 			t.Fatalf("%+v", ge)
 		}
 	} else {
-		t.Fatalf("Threre should be error, when table name duplicates.")
+		t.Fatalf(("Threre should be error, when table name duplicates."))
```
Comment (mresh-sigma) [inline server/server_test.go]: do we need two parentheses?

Comment (mresh-sigma) [reply server/server_test.go]: typo: Threre -> There

File: server/server_test.go [hunk 2]
```diff
@@ -806,7 +921,7 @@ func TestDuplicateTableWithSchema(t *testing.T) {
...
```
Comment (mresh-sigma) [inline server/server_test.go]: typo: Threre -> There

File: server/server_test.go (hunk 1 above)
Comment (denys-klymenko-sigma) [reply server/server_test.go]: wow, good catch, thank you

**Comment by ManuelPoppeRichter** [comment]
@denys-klymenko-sigma It says files changed, 5,500+, is that normal?

**Comment by denys-klymenko-sigma** [comment]
> @denys-klymenko-sigma It says files changed, 5,500+, is that normal?

Yeah, for some reason(extra stability I guess. @diogogmt could probably shed some light to the reasons) we vendor all dependencies, sooo it's a lot of files :) 

For convenience all "real" changes are in separate commit

File: go.mod [hunk 3]
```diff
@@ -1,88 +1,100 @@
 module github.com/goccy/bigquery-emulator
 
-go 1.20
+go 1.22
 
 require (
-	cloud.google.com/go/bigquery v1.51.0
-	cloud.google.com/go/storage v1.29.0
+	cloud.google.com/go/bigquery v1.60.0
```
Comment (zhehlovvalentyn) [inline go.mod]: should we remove this after the merge of the PR?
https://github.com/blockthrough/up-marble/blob/master/go.mod#L279-L280

File: internal/contentdata/repository.go [hunk 4]
```diff
@@ -367,34 +370,58 @@ func (r *Repository) AddTableData(ctx context.Context, tx *connection.Tx, projec
...
+	if err != nil {
 		return err
 	}
+
+	for _, data := range table.Data {
+		values := make([]interface{}, 0, len(table.Columns))
+
+		for _, column := range columns {
+			if value, found := data[column.Name]; found {
+				isTimestampColumn := column.Type == types.TIMESTAMP
```
Comment (zhehlovvalentyn) [inline internal/contentdata/repository.go]: do we need this var? we use it only once

File: server/server.go [hunk 5]
```diff
@@ -212,7 +214,7 @@ func (s *Server) Serve(ctx context.Context, httpAddr, grpcAddr string) error {
 	httpServer := &http.Server{
 		Handler:      s.Handler,
 		Addr:         httpAddr,
-		WriteTimeout: 15 * time.Second,
+		WriteTimeout: 5 * time.Minute,
```
Comment (zhehlovvalentyn) [inline server/server.go]: 5 minutes? not against it but why so much?

**Review by zhehlovvalentyn — APPROVED** [review]

File: server/server.go (hunk 5 above)
Comment (denys-klymenko-sigma) [reply server/server.go]: https://github.com/goccy/bigquery-emulator/commit/65eafe1674ce5a97bfe5a6b87a503308b267f8a7 
apparently in cases when timeout happens job is created anyway and it's hard to recover from that state

File: internal/contentdata/repository.go (hunk 4 above)
Comment (denys-klymenko-sigma) [reply internal/contentdata/repository.go]: removed

File: go.mod (hunk 3 above)
Comment (denys-klymenko-sigma) [reply go.mod]: removed it in up-mable and everything works fine

**Pull request merged**